### ISSUE TITLE
NVDA Alpha fails to clean its system tray icon after restart

### ISSUE THREAD

**Issue by lukaszgo1** [nvaccess/nvda#12238]
### Steps to reproduce:
Start NVDA Alpha, restart it ten times in a row and inspect system tray.
### Actual behavior:
There are 10 NVDA icons in system tray - 9 of them are dead and would disappear when clicked.
### Expected behavior:
There should be just one icon for the currently running instance of NVDA.

#### NVDA installed/portable/running from source:
Portable
#### NVDA version:
NVDA version alpha-22160,656ecb5f
#### Windows version:
Windows 7 X64
#### Name and version of other software in use when reproducing the issue:
None
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
2020.4 works correctly some older Alphas works also.

#### If add-ons are disabled, is your problem still occurring?
Yes
#### Did you try to run the COM registry fixing tool in NVDA menu / tools?
NO

cc @seanbudd

**Comment by lukaszgo1** [comment]
Probably a regression from 8665526b573affa35d9d89b0650972482e45dc4c as of this commit were no  longer destroying our system tray icon.

**Comment by XLTechie** [comment]
Related to #11553

**Comment by Brian1Gaff** [comment]
I noticed this as well, as you say this used to happen many versions ago as 
well.
 Brian

***@***.***
Sent via blueyonder.
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
Newsgroup monitored: alt.comp.blind-users
----- Original Message -----

**Comment by trypolis464** [comment]
This never actually stopped happening for me. It will be good to see it fixed. It might be a WXPython issue, not sure though, but TW Blue does it too.

**Comment by seanbudd** [comment]
I reproduced this with `alpha-22119,8412adbd` so this isn't a regression from `8665526`.

**Comment by seanbudd** [comment]
this is also reproduced on `2020.4`. I haven't been able to find a way to consistently fix this and considering it's not a regression I'm going to take this off the milestone for 2021.1

**Comment by CyrilleB79** [comment]
> this is also reproduced on `2020.4`. I haven't been able to find a way to consistently fix this and considering it's not a regression I'm going to take this off the milestone for 2021.1

According to my tests, the issue seems to occur on NVDA 2020.4 only occasionally and after having worked several hours with NVDA. On the contrary, on last alpha (nvda_snapshot_alpha-22295,49a85788), the issue seems to be reproducible systematically.
Tests performed with NVDA 2020.4 installed version and NVDA alpha portable.
So IMO, there is a regression in this dev cycle.

In any case, the severity of this bug remains low of course even if the number of occurrences has increased.

**Comment by seanbudd** [comment]
Reopened due to #12326

**Comment by lukaszgo1** [comment]
I'm afraid this is not fixed for me with the initial STR. Restarting latest Alpha of NVDA ten times in a row leaves nine dead NVDA icons in the system tray whereas doing the same with 2020.4 only one icon is present. At first I thought this may be caused by wxPython upgrade and therefore be outside of our control but performing initial STR with the build from #12075 i.e. just after wxPython 4.1.1 has been introduced does not reproduce this either. Can this be reopened please?

**Comment by lukaszgo1** [comment]
I'm sorry - it turns out there is no binary build with this issue fixed. Testing from sources I can confirm this issue no longer occurs.
